src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -12906,5 +12959,20 @@ private new struct ResetPoint
             node = this.AddTrailingSkippedSyntax(node, trailingTrash.Node);
             return node;
         }
+
+        private SyntaxToken MergeTokens(SyntaxToken s1, SyntaxToken s2, SyntaxKind kind)
+        {
+            if (s1.GetTrailingTriviaWidth() == 0 && s2.GetLeadingTriviaWidth() == 0)
```

Comment:
>s1.GetTrailingTriviaWidth() == 0 && s2.GetLeadingTriviaWidth() == 0 [](start = 16, length = 67)

Are we testing both cases where this fails? (When `s1.GetTrailingTriviaWidth() > 0` and when `s1.GetTrailingTriviaWidth() == 0 && s2.GetLeadingTriviaWidth() > 0`?)

For instance:
```
static void F(object arg!
    != null);
```

Reply:
Here are some of the cases covered:
```
    void M0(string name !!=null) { }
    void M1(string name! !=null) { }
    void M2(string name!!= null) { }
    void M3(string name ! !=null) { }
    void M4(string name ! ! =null) { }
    void M5(string name! ! =null) { }
```